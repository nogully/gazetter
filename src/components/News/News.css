.News {
  display: flex;
  flex-wrap: wrap;
  text-align: center;
  max-width: 1000px;
  flex-direction: row;    
  justify-content: center;
  margin: 0;
}

.tweet {
  border-bottom: 1px solid #00aced;
  padding: 20px;
  margin: 10px;
  width: 300px;
}

p {
  font-family: 'Merriweather', serif;
  margin: 0px;
  line-height: 1.5em;
  text-align: left;
  }

h5 {
  font-family: 'Merriweather', serif;
  color: #31424B;
  margin: 0px;
  font-weight: 700;
  font-variant: small-caps;
  text-align: right;
}

img {
  width: 300px;
}