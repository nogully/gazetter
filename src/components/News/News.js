import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom'
import './News.css'
// import PropTypes from 'prop-types';

export class News extends Component {

  tweetCards = () => {
    const { tweets } = this.props;
    return tweets.map( tweet => {
      return <p>{tweet.text}</p>
    })
     
  }

  render() {
    return (
      <div>
      <h1> News </h1>
      {this.tweetCards()}
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  tweets: store.tweets
})

export default withRouter(connect(mapStateToProps, null)(News))