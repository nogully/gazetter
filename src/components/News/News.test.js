/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { News, mapStateToProps } from './News';

describe('News', () => {
  it('should match its snapshot', () => {
    const mockTweets = [
              { text: 'Why tho, Jhun?', 
                user: { display_name: 'noragully'}, 
                id: 1234 }, 
              { text: 'What\'s good, fam?', 
                user: { display_name: 'yung-jhun'}, 
                id: 5678 } 
            ]
    const wrapper = shallow(<News tweets={ mockTweets } />)
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should define the tweets props for the News container via Redux', () => {
      const mockTweets = [
              { text: 'Why tho, Jhun?', 
                user: { display_name: 'noragully'}, 
                id: 1234 }, 
              { text: 'What\'s good, fam?', 
                user: { display_name: 'yung-jhun'}, 
                id: 5678 } 
            ]
      const mockStore = { tweets: mockTweets };
      const expectedProps = { tweets: [
              { text: 'Why tho, Jhun?', 
                user: { display_name: 'noragully'}, 
                id: 1234 }, 
              { text: 'What\'s good, fam?', 
                user: { display_name: 'yung-jhun'}, 
                id: 5678 } 
            ] }
      const mapped = mapStateToProps(mockStore);
      expect(mapped).toEqual(expectedProps)
    })
  })

})