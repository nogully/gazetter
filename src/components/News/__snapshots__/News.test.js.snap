// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News should match its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <News
    tweets={
        Array [
            Object {
              "id": 1234,
              "text": "Why tho, Jhun?",
              "user": Object {
                "display_name": "noragully",
              },
            },
            Object {
              "id": 5678,
              "text": "What's good, fam?",
              "user": Object {
                "display_name": "yung-jhun",
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Article
          tweet={
                    Object {
                              "id": 1234,
                              "text": "Why tho, Jhun?",
                              "user": Object {
                                "display_name": "noragully",
                              },
                            }
          }
/>,
        <Article
          tweet={
                    Object {
                              "id": 5678,
                              "text": "What's good, fam?",
                              "user": Object {
                                "display_name": "yung-jhun",
                              },
                            }
          }
/>,
      ],
      "className": "News",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1234",
        "nodeType": "function",
        "props": Object {
          "tweet": Object {
            "id": 1234,
            "text": "Why tho, Jhun?",
            "user": Object {
              "display_name": "noragully",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5678",
        "nodeType": "function",
        "props": Object {
          "tweet": Object {
            "id": 5678,
            "text": "What's good, fam?",
            "user": Object {
              "display_name": "yung-jhun",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Article
            tweet={
                        Object {
                                    "id": 1234,
                                    "text": "Why tho, Jhun?",
                                    "user": Object {
                                      "display_name": "noragully",
                                    },
                                  }
            }
/>,
          <Article
            tweet={
                        Object {
                                    "id": 5678,
                                    "text": "What's good, fam?",
                                    "user": Object {
                                      "display_name": "yung-jhun",
                                    },
                                  }
            }
/>,
        ],
        "className": "News",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1234",
          "nodeType": "function",
          "props": Object {
            "tweet": Object {
              "id": 1234,
              "text": "Why tho, Jhun?",
              "user": Object {
                "display_name": "noragully",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5678",
          "nodeType": "function",
          "props": Object {
            "tweet": Object {
              "id": 5678,
              "text": "What's good, fam?",
              "user": Object {
                "display_name": "yung-jhun",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
